pkg/subctl/cmd/gather.go

```diff
@@ -125,6 +129,24 @@ func gatherDataByCluster(restConfig restConfig) {
 	info.DynClient, info.ClientSet, err = getClients(restConfig.config)
 	exitOnError("Error getting client %s", err)
 
+	resourceSubmariners := schema.GroupVersionResource{
+		Group:    submarinerOp.SchemeGroupVersion.Group,
+		Version:  submarinerOp.SchemeGroupVersion.Version,
+		Resource: "submariners",
+	}
+
+	var submariner submarinerOp.Submariner
+
+	resourceSubm, err := info.DynClient.Resource(resourceSubmariners).Namespace(submarinerNamespace).Get("submariner", metav1.GetOptions{})
+	if err != nil {
+		exitOnError(fmt.Sprintf("Failed to get submariner resource %q", err), err)
```

Comment:
This should not `exitOnError` but print warning message that subm resource is not found. Otherwise, it skips collecting other resources from cluster where subm is not installed eg broker

```
$ ./../submariner-operator/bin/subctl gather module=connectivity
**Gathering information from cluster "cluster2"**
 ✓ Gathering connectivity logs 
....
 ✓ Found 1 deployments by label selector "app=submariner-lighthouse-coredns" in namespace "submariner-operator"
**Gathering information from cluster "cluster3"**
 ✓ Gathering operator logs 
...
**Gathering information from cluster "cluster1"**
panic: runtime error: invalid memory address or nil pointer dereference
[signal] SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14b553a]
```

For me, cluster1 is broker cluster and it should still collect information like `Endpoints` and  `Clusters`.

I understand that broker resources are collected while iterating over dataclusters but that is something that developer knows. User gets confused with the `exitOnError` response.

Reply:
There is a `getSubmarinerResource` function that does this. However we shouldn't get it here and error if not found. In addition to what @Jaanki mentioned, In a service-discovery-only install there is no `Submariner` resource. So we should obtain it where needed and warn/skip appropriately if not found.

Reply:
Oh good point

Reply:
I'll remove this error. This code will be reworked once https://github.com/submariner-io/submariner-operator/pull/1252 and subsequent patches are merged. For some of LH gathering, we need serviceDiscovery CR.